internal/mode/static/telemetry/collector.go

```diff
@@ -378,3 +402,110 @@ func collectClusterInformation(ctx context.Context, k8sClient client.Reader) (cl
 
 	return clusterInfo, nil
 }
+
+type sfContextDirective struct {
+	context   string
+	directive string
+}
+
+func collectSnippetsFilterSnippetsInfo(graphGetter GraphGetter) ([]string, []int64, error) {
+	g := graphGetter.GetLatestGraph()
+	if g == nil {
+		return nil, nil, errors.New("latest graph cannot be nil")
+	}
+
+	contextDirectiveMap := make(map[sfContextDirective]int)
+
+	for name := range g.SnippetsFilters {
+		sf := g.SnippetsFilters[name]
+		if sf == nil {
+			continue
+		}
+
+		for nginxContext := range sf.Snippets {
+			snippetValue := sf.Snippets[nginxContext]
+
+			var parsedContext string
+			switch nginxContext {
+			case ngfAPI.NginxContextMain:
+				parsedContext = "main"
+			case ngfAPI.NginxContextHTTP:
+				parsedContext = "http"
+			case ngfAPI.NginxContextHTTPServer:
+				parsedContext = "server"
+			case ngfAPI.NginxContextHTTPServerLocation:
+				parsedContext = "location"
+			default:
+				parsedContext = "unknown"
+			}
+
+			directives := parseSnippetValueIntoDirectives(snippetValue)
+
+			for _, directive := range directives {
+				contextDirective := sfContextDirective{
+					context:   parsedContext,
+					directive: directive,
+				}
+
+				contextDirectiveMap[contextDirective]++
+			}
+		}
+	}
+
+	contextDirectiveList, countList := parseContextDirectiveMapIntoLists(contextDirectiveMap)
+
+	return contextDirectiveList, countList, nil
+}
+
+func parseSnippetValueIntoDirectives(snippetValue string) []string {
+	separatedDirectives := strings.Split(snippetValue, ";")
+	directives := make([]string, 0, len(separatedDirectives))
+
+	for _, directive := range separatedDirectives {
+		// the strings.TrimSpace is needed in the case of multi-line NGINX Snippet values
+		directive = strings.Split(strings.TrimSpace(directive), " ")[0]
```

Comment:
Does this also handle cleaning out newlines? If newlines are represented as `\n` in the literal string, then you could use `strings.ReplaceAll(str, "\n", "")`

Reply:
Hm, it does clear our the leading and trailing newline representations. And since I split on the `;` I assumed that there would be no wonky newline characters in the middle of an nginx directive value. I could only see them being between different nginx directives.


Do you think I should worry about that? For example:

`aio \n \n on;`

Reply:
If TrimSpace does clear out newline characters, then that's fine. Just reading the function description, it only mentions whitespace